dali/python/nvidia/dali/pipeline.py

```diff
@@ -777,10 +783,52 @@ def contains_nested_datanode(nested):
 
         self._ops = _collect_ops(list(outputs) + self._sinks)
         self._graph_outputs = outputs
+        self._rename_foreign_ops()
         self._setup_input_callbacks()
         self._disable_pruned_external_source_instances()
         self._py_graph_built = True
 
+    def _rename_foreign_ops(self):
```

Comment:
How about adding a comment with an example illustrating how this renaming works?

Reply:
Done.